src/main/scala/com/microsoft/hyperspace/index/dataskipping/sketch/BloomFilterSketch.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (2021) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.index.dataskipping.sketch
+
+import org.apache.spark.sql.Column
+import org.apache.spark.sql.functions.column
+
+/**
+ * Sketch based on a bloom filter for a given column.
+ *
+ * Being a probabilistic structure, it is more efficient in terms of the index
+ * data size than [[ValueListSketch]] if the number of distinct values for the
+ * column is large, but can be less efficient in terms of query optimization
+ * than [[ValueListSketch]] due to false positives.
+ *
+ * Users can specify the target false positive rate and the expected number of
+ * distinct values per file. These variables determine the size of the bloom
+ * filters and thus the size of the index data.
```

Comment:
TODO: We can do this semi-automatically.

The expected number of distinct values per file can be directly calculated in Spark. The problem is there is no way to specify this differently for each file in the current implementation, as aggregation functions don't know about the current grouping key (= input_file_name() for our case). As a workaround, we can use the average value, although it won't be ideal for files varying greatly in size or distribution of key values.

For a better user experience, instead of specifying FPP, users should be able to specify the target size of the index. Then we can choose FPP based on the number of disticted values. If the calculated FPP is too high, users should be warned about it.

Reply:
Then we can make a ratio-based config and set a default value for it - for example, aim for 0.1% (default, configurable) of the source data size.